packages/multicast_dns/test/decode_test.dart

```diff
@@ -192,6 +192,10 @@ void testBadPackages() {
       }
     }
   });
+
+  test('Detects cyclic pointers and returns null', () {
+    expect(decodeMDnsResponse(cycle), isNull);
```

Comment:
I verified that this is actually detecting a cycle using the debugger. We could refactor `decodeMDnsResponse` to return some sort of ADT that can indicate a reason for failure instead of always returning null. Then, tests could make sure that null is being returned by the expected reason. However, I want to make sure you are OK with this cycle-detection approach before I invest in that effort.